src/pspm_write_channel.m

```diff
@@ -95,17 +95,21 @@
         if isfield(newdata{i}, 'data') && isfield(newdata{i}, 'header')
             d = newdata{i}.data;
             [h,w] = size(d);
-            if w ~= 1
+            if h==0 && w==0
+              newdata{i}.marker = 'empty';
+            else
+              if w ~= 1
                 if h == 1
-                    warning('ID:invalid_data_structure', ['Passed struct (%i) seems to have the wrong ', ...
-                        'orientation. Trying to transpose...'], i);
-                    d = d';
-                    newdata{i}.data = d;
+                  warning('ID:invalid_data_structure', ['Passed struct (%i) seems to have the wrong ', ...
+                    'orientation. Trying to transpose...'], i);
+                  d = d';
+                  newdata{i}.data = d;
                 else
-                    warning('ID:invalid_data_structure', ...
-                        'Passed struct (%i) seems to have the wrong format.', i);
-                    return;
+                  warning('ID:invalid_data_structure', ...

```

Comment:
The `^M` is caused by the difference between CRLF and LF. I will update that after the approval of this pull request, because updating will highlight too many lines and confuse reviewers.